lib/drivers/veres/index.js

```diff
@@ -321,27 +381,41 @@ api.notes = async options => {
   // return await release();
 };
 
-/* eslint-disable-next-line no-unused-vars */
-api['authn-rotate'] = async options => {
-  // TODO: remove old, add new, track old id as used, revoke old
-  throw new Error('authn-rotate not implemented');
-};
+/**
+ * @param options
+ * @param {string} options.keyId - Id of the existing key to rotate.
+ * @param {string} [options.mode='test'] - Ledger mode (test/live/dev), if
+ *   applicable.
+ * @param {string} [options.hostname] - Ledger node hostname override
+ * @param {string} [options.type='veres'] - DID type / method.
+ *
+ * @returns {Promise}
+ */
+api['rotate'] = async options => {
+  const {mode, hostname, type, keyId} = options;
+  const veresDriver = driver({mode, hostname});
 
-/* eslint-disable-next-line no-unused-vars */
-api['ocap-add'] = async options => {
-  throw new Error('ocap-add not implemented');
-};
+  const did = keyId.split('#')[0];
 
-/* eslint-disable-next-line no-unused-vars */
-api['ocap-revoke'] = async options => {
-  throw new Error('ocap-add not implemented');
+  const {status, didDocument, meta} =
+    await loadDidDocument({did, veresDriver, mode, type});
+
+  await didDocument.rotateKey({id: keyId});
+
+  await saveDidDocument({didDocument, status, type, mode, veresDriver});
+  logger._log(options, 'Saved updated document.');
+
+  // Update was successful, save new private key to key store
+  await keyStore().put(did, await didDocument.exportKeys());
+  logger._log(options, 'Saved new private key to local key store.');
+
+  // TODO: Update metadata (to record rotate event)
```

Comment:
Does this need to be completed?

Reply:
I don't think it's urgent (or at least, I'm not quite clear on what is the proper json-ld way to do an event log type thing)?